{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    },
    "gpuClass": "standard"
  },
  "cells": [
    {
      "cell_type": "code",
      "source": [
        "!git clone https://github.com/Zedai/PyTorch-WaveNet.git"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Y9f1Ff48py60",
        "outputId": "b547b92a-c193-4ecd-a648-d550b5529b9e"
      },
      "execution_count": 2,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Cloning into 'PyTorch-WaveNet'...\n",
            "remote: Enumerating objects: 54, done.\u001b[K\n",
            "remote: Counting objects: 100% (54/54), done.\u001b[K\n",
            "remote: Compressing objects: 100% (45/45), done.\u001b[K\n",
            "remote: Total 54 (delta 14), reused 3 (delta 0), pack-reused 0\u001b[K\n",
            "Unpacking objects: 100% (54/54), done.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install -r PyTorch-WaveNet/requirements_gpu.txt\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Mq_0qMeGrreW",
        "outputId": "8fe02f09-a90d-4dfd-e95b-9b53a43dfdcd"
      },
      "execution_count": 3,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n",
            "Requirement already satisfied: librosa>=0.5 in /usr/local/lib/python3.8/dist-packages (from -r PyTorch-WaveNet/requirements_gpu.txt (line 1)) (0.8.1)\n",
            "Collecting tensorflow-gpu>=1.0.0\n",
            "  Downloading tensorflow_gpu-2.11.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (588.3 MB)\n",
            "\u001b[K     |████████████████████████████████| 588.3 MB 20 kB/s \n",
            "\u001b[?25hRequirement already satisfied: decorator>=3.0.0 in /usr/local/lib/python3.8/dist-packages (from librosa>=0.5->-r PyTorch-WaveNet/requirements_gpu.txt (line 1)) (4.4.2)\n",
            "Requirement already satisfied: numpy>=1.15.0 in /usr/local/lib/python3.8/dist-packages (from librosa>=0.5->-r PyTorch-WaveNet/requirements_gpu.txt (line 1)) (1.21.6)\n",
            "Requirement already satisfied: audioread>=2.0.0 in /usr/local/lib/python3.8/dist-packages (from librosa>=0.5->-r PyTorch-WaveNet/requirements_gpu.txt (line 1)) (3.0.0)\n",
            "Requirement already satisfied: pooch>=1.0 in /usr/local/lib/python3.8/dist-packages (from librosa>=0.5->-r PyTorch-WaveNet/requirements_gpu.txt (line 1)) (1.6.0)\n",
            "Requirement already satisfied: resampy>=0.2.2 in /usr/local/lib/python3.8/dist-packages (from librosa>=0.5->-r PyTorch-WaveNet/requirements_gpu.txt (line 1)) (0.4.2)\n",
            "Requirement already satisfied: soundfile>=0.10.2 in /usr/local/lib/python3.8/dist-packages (from librosa>=0.5->-r PyTorch-WaveNet/requirements_gpu.txt (line 1)) (0.11.0)\n",
            "Requirement already satisfied: numba>=0.43.0 in /usr/local/lib/python3.8/dist-packages (from librosa>=0.5->-r PyTorch-WaveNet/requirements_gpu.txt (line 1)) (0.56.4)\n",
            "Requirement already satisfied: joblib>=0.14 in /usr/local/lib/python3.8/dist-packages (from librosa>=0.5->-r PyTorch-WaveNet/requirements_gpu.txt (line 1)) (1.2.0)\n",
            "Requirement already satisfied: scikit-learn!=0.19.0,>=0.14.0 in /usr/local/lib/python3.8/dist-packages (from librosa>=0.5->-r PyTorch-WaveNet/requirements_gpu.txt (line 1)) (1.0.2)\n",
            "Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.8/dist-packages (from librosa>=0.5->-r PyTorch-WaveNet/requirements_gpu.txt (line 1)) (21.3)\n",
            "Requirement already satisfied: scipy>=1.0.0 in /usr/local/lib/python3.8/dist-packages (from librosa>=0.5->-r PyTorch-WaveNet/requirements_gpu.txt (line 1)) (1.7.3)\n",
            "Requirement already satisfied: protobuf<3.20,>=3.9.2 in /usr/local/lib/python3.8/dist-packages (from tensorflow-gpu>=1.0.0->-r PyTorch-WaveNet/requirements_gpu.txt (line 2)) (3.19.6)\n",
            "Requirement already satisfied: termcolor>=1.1.0 in /usr/local/lib/python3.8/dist-packages (from tensorflow-gpu>=1.0.0->-r PyTorch-WaveNet/requirements_gpu.txt (line 2)) (2.1.1)\n",
            "Collecting tensorboard<2.12,>=2.11\n",
            "  Downloading tensorboard-2.11.0-py3-none-any.whl (6.0 MB)\n",
            "\u001b[K     |████████████████████████████████| 6.0 MB 6.5 MB/s \n",
            "\u001b[?25hRequirement already satisfied: absl-py>=1.0.0 in /usr/local/lib/python3.8/dist-packages (from tensorflow-gpu>=1.0.0->-r PyTorch-WaveNet/requirements_gpu.txt (line 2)) (1.3.0)\n",
            "Requirement already satisfied: astunparse>=1.6.0 in /usr/local/lib/python3.8/dist-packages (from tensorflow-gpu>=1.0.0->-r PyTorch-WaveNet/requirements_gpu.txt (line 2)) (1.6.3)\n",
            "Requirement already satisfied: six>=1.12.0 in /usr/local/lib/python3.8/dist-packages (from tensorflow-gpu>=1.0.0->-r PyTorch-WaveNet/requirements_gpu.txt (line 2)) (1.15.0)\n",
            "Requirement already satisfied: typing-extensions>=3.6.6 in /usr/local/lib/python3.8/dist-packages (from tensorflow-gpu>=1.0.0->-r PyTorch-WaveNet/requirements_gpu.txt (line 2)) (4.4.0)\n",
            "Requirement already satisfied: gast<=0.4.0,>=0.2.1 in /usr/local/lib/python3.8/dist-packages (from tensorflow-gpu>=1.0.0->-r PyTorch-WaveNet/requirements_gpu.txt (line 2)) (0.4.0)\n",
            "Requirement already satisfied: tensorflow-io-gcs-filesystem>=0.23.1 in /usr/local/lib/python3.8/dist-packages (from tensorflow-gpu>=1.0.0->-r PyTorch-WaveNet/requirements_gpu.txt (line 2)) (0.28.0)\n",
            "Collecting flatbuffers>=2.0\n",
            "  Downloading flatbuffers-22.12.6-py2.py3-none-any.whl (26 kB)\n",
            "Collecting keras<2.12,>=2.11.0\n",
            "  Downloading keras-2.11.0-py2.py3-none-any.whl (1.7 MB)\n",
            "\u001b[K     |████████████████████████████████| 1.7 MB 29.7 MB/s \n",
            "\u001b[?25hCollecting tensorflow-estimator<2.12,>=2.11.0\n",
            "  Downloading tensorflow_estimator-2.11.0-py2.py3-none-any.whl (439 kB)\n",
            "\u001b[K     |████████████████████████████████| 439 kB 63.9 MB/s \n",
            "\u001b[?25hRequirement already satisfied: opt-einsum>=2.3.2 in /usr/local/lib/python3.8/dist-packages (from tensorflow-gpu>=1.0.0->-r PyTorch-WaveNet/requirements_gpu.txt (line 2)) (3.3.0)\n",
            "Requirement already satisfied: h5py>=2.9.0 in /usr/local/lib/python3.8/dist-packages (from tensorflow-gpu>=1.0.0->-r PyTorch-WaveNet/requirements_gpu.txt (line 2)) (3.1.0)\n",
            "Requirement already satisfied: setuptools in /usr/local/lib/python3.8/dist-packages (from tensorflow-gpu>=1.0.0->-r PyTorch-WaveNet/requirements_gpu.txt (line 2)) (57.4.0)\n",
            "Requirement already satisfied: grpcio<2.0,>=1.24.3 in /usr/local/lib/python3.8/dist-packages (from tensorflow-gpu>=1.0.0->-r PyTorch-WaveNet/requirements_gpu.txt (line 2)) (1.51.1)\n",
            "Requirement already satisfied: libclang>=13.0.0 in /usr/local/lib/python3.8/dist-packages (from tensorflow-gpu>=1.0.0->-r PyTorch-WaveNet/requirements_gpu.txt (line 2)) (14.0.6)\n",
            "Requirement already satisfied: wrapt>=1.11.0 in /usr/local/lib/python3.8/dist-packages (from tensorflow-gpu>=1.0.0->-r PyTorch-WaveNet/requirements_gpu.txt (line 2)) (1.14.1)\n",
            "Requirement already satisfied: google-pasta>=0.1.1 in /usr/local/lib/python3.8/dist-packages (from tensorflow-gpu>=1.0.0->-r PyTorch-WaveNet/requirements_gpu.txt (line 2)) (0.2.0)\n",
            "Requirement already satisfied: wheel<1.0,>=0.23.0 in /usr/local/lib/python3.8/dist-packages (from astunparse>=1.6.0->tensorflow-gpu>=1.0.0->-r PyTorch-WaveNet/requirements_gpu.txt (line 2)) (0.38.4)\n",
            "Requirement already satisfied: llvmlite<0.40,>=0.39.0dev0 in /usr/local/lib/python3.8/dist-packages (from numba>=0.43.0->librosa>=0.5->-r PyTorch-WaveNet/requirements_gpu.txt (line 1)) (0.39.1)\n",
            "Requirement already satisfied: importlib-metadata in /usr/local/lib/python3.8/dist-packages (from numba>=0.43.0->librosa>=0.5->-r PyTorch-WaveNet/requirements_gpu.txt (line 1)) (5.1.0)\n",
            "Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in /usr/local/lib/python3.8/dist-packages (from packaging>=20.0->librosa>=0.5->-r PyTorch-WaveNet/requirements_gpu.txt (line 1)) (3.0.9)\n",
            "Requirement already satisfied: requests>=2.19.0 in /usr/local/lib/python3.8/dist-packages (from pooch>=1.0->librosa>=0.5->-r PyTorch-WaveNet/requirements_gpu.txt (line 1)) (2.23.0)\n",
            "Requirement already satisfied: appdirs>=1.3.0 in /usr/local/lib/python3.8/dist-packages (from pooch>=1.0->librosa>=0.5->-r PyTorch-WaveNet/requirements_gpu.txt (line 1)) (1.4.4)\n",
            "Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /usr/local/lib/python3.8/dist-packages (from requests>=2.19.0->pooch>=1.0->librosa>=0.5->-r PyTorch-WaveNet/requirements_gpu.txt (line 1)) (1.24.3)\n",
            "Requirement already satisfied: chardet<4,>=3.0.2 in /usr/local/lib/python3.8/dist-packages (from requests>=2.19.0->pooch>=1.0->librosa>=0.5->-r PyTorch-WaveNet/requirements_gpu.txt (line 1)) (3.0.4)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.8/dist-packages (from requests>=2.19.0->pooch>=1.0->librosa>=0.5->-r PyTorch-WaveNet/requirements_gpu.txt (line 1)) (2022.12.7)\n",
            "Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.8/dist-packages (from requests>=2.19.0->pooch>=1.0->librosa>=0.5->-r PyTorch-WaveNet/requirements_gpu.txt (line 1)) (2.10)\n",
            "Requirement already satisfied: threadpoolctl>=2.0.0 in /usr/local/lib/python3.8/dist-packages (from scikit-learn!=0.19.0,>=0.14.0->librosa>=0.5->-r PyTorch-WaveNet/requirements_gpu.txt (line 1)) (3.1.0)\n",
            "Requirement already satisfied: cffi>=1.0 in /usr/local/lib/python3.8/dist-packages (from soundfile>=0.10.2->librosa>=0.5->-r PyTorch-WaveNet/requirements_gpu.txt (line 1)) (1.15.1)\n",
            "Requirement already satisfied: pycparser in /usr/local/lib/python3.8/dist-packages (from cffi>=1.0->soundfile>=0.10.2->librosa>=0.5->-r PyTorch-WaveNet/requirements_gpu.txt (line 1)) (2.21)\n",
            "Requirement already satisfied: werkzeug>=1.0.1 in /usr/local/lib/python3.8/dist-packages (from tensorboard<2.12,>=2.11->tensorflow-gpu>=1.0.0->-r PyTorch-WaveNet/requirements_gpu.txt (line 2)) (1.0.1)\n",
            "Requirement already satisfied: google-auth-oauthlib<0.5,>=0.4.1 in /usr/local/lib/python3.8/dist-packages (from tensorboard<2.12,>=2.11->tensorflow-gpu>=1.0.0->-r PyTorch-WaveNet/requirements_gpu.txt (line 2)) (0.4.6)\n",
            "Requirement already satisfied: tensorboard-data-server<0.7.0,>=0.6.0 in /usr/local/lib/python3.8/dist-packages (from tensorboard<2.12,>=2.11->tensorflow-gpu>=1.0.0->-r PyTorch-WaveNet/requirements_gpu.txt (line 2)) (0.6.1)\n",
            "Requirement already satisfied: tensorboard-plugin-wit>=1.6.0 in /usr/local/lib/python3.8/dist-packages (from tensorboard<2.12,>=2.11->tensorflow-gpu>=1.0.0->-r PyTorch-WaveNet/requirements_gpu.txt (line 2)) (1.8.1)\n",
            "Requirement already satisfied: google-auth<3,>=1.6.3 in /usr/local/lib/python3.8/dist-packages (from tensorboard<2.12,>=2.11->tensorflow-gpu>=1.0.0->-r PyTorch-WaveNet/requirements_gpu.txt (line 2)) (2.15.0)\n",
            "Requirement already satisfied: markdown>=2.6.8 in /usr/local/lib/python3.8/dist-packages (from tensorboard<2.12,>=2.11->tensorflow-gpu>=1.0.0->-r PyTorch-WaveNet/requirements_gpu.txt (line 2)) (3.4.1)\n",
            "Requirement already satisfied: rsa<5,>=3.1.4 in /usr/local/lib/python3.8/dist-packages (from google-auth<3,>=1.6.3->tensorboard<2.12,>=2.11->tensorflow-gpu>=1.0.0->-r PyTorch-WaveNet/requirements_gpu.txt (line 2)) (4.9)\n",
            "Requirement already satisfied: pyasn1-modules>=0.2.1 in /usr/local/lib/python3.8/dist-packages (from google-auth<3,>=1.6.3->tensorboard<2.12,>=2.11->tensorflow-gpu>=1.0.0->-r PyTorch-WaveNet/requirements_gpu.txt (line 2)) (0.2.8)\n",
            "Requirement already satisfied: cachetools<6.0,>=2.0.0 in /usr/local/lib/python3.8/dist-packages (from google-auth<3,>=1.6.3->tensorboard<2.12,>=2.11->tensorflow-gpu>=1.0.0->-r PyTorch-WaveNet/requirements_gpu.txt (line 2)) (5.2.0)\n",
            "Requirement already satisfied: requests-oauthlib>=0.7.0 in /usr/local/lib/python3.8/dist-packages (from google-auth-oauthlib<0.5,>=0.4.1->tensorboard<2.12,>=2.11->tensorflow-gpu>=1.0.0->-r PyTorch-WaveNet/requirements_gpu.txt (line 2)) (1.3.1)\n",
            "Requirement already satisfied: zipp>=0.5 in /usr/local/lib/python3.8/dist-packages (from importlib-metadata->numba>=0.43.0->librosa>=0.5->-r PyTorch-WaveNet/requirements_gpu.txt (line 1)) (3.11.0)\n",
            "Requirement already satisfied: pyasn1<0.5.0,>=0.4.6 in /usr/local/lib/python3.8/dist-packages (from pyasn1-modules>=0.2.1->google-auth<3,>=1.6.3->tensorboard<2.12,>=2.11->tensorflow-gpu>=1.0.0->-r PyTorch-WaveNet/requirements_gpu.txt (line 2)) (0.4.8)\n",
            "Requirement already satisfied: oauthlib>=3.0.0 in /usr/local/lib/python3.8/dist-packages (from requests-oauthlib>=0.7.0->google-auth-oauthlib<0.5,>=0.4.1->tensorboard<2.12,>=2.11->tensorflow-gpu>=1.0.0->-r PyTorch-WaveNet/requirements_gpu.txt (line 2)) (3.2.2)\n",
            "Installing collected packages: tensorflow-estimator, tensorboard, keras, flatbuffers, tensorflow-gpu\n",
            "  Attempting uninstall: tensorflow-estimator\n",
            "    Found existing installation: tensorflow-estimator 2.9.0\n",
            "    Uninstalling tensorflow-estimator-2.9.0:\n",
            "      Successfully uninstalled tensorflow-estimator-2.9.0\n",
            "  Attempting uninstall: tensorboard\n",
            "    Found existing installation: tensorboard 2.9.1\n",
            "    Uninstalling tensorboard-2.9.1:\n",
            "      Successfully uninstalled tensorboard-2.9.1\n",
            "  Attempting uninstall: keras\n",
            "    Found existing installation: keras 2.9.0\n",
            "    Uninstalling keras-2.9.0:\n",
            "      Successfully uninstalled keras-2.9.0\n",
            "  Attempting uninstall: flatbuffers\n",
            "    Found existing installation: flatbuffers 1.12\n",
            "    Uninstalling flatbuffers-1.12:\n",
            "      Successfully uninstalled flatbuffers-1.12\n",
            "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
            "tensorflow 2.9.2 requires flatbuffers<2,>=1.12, but you have flatbuffers 22.12.6 which is incompatible.\n",
            "tensorflow 2.9.2 requires keras<2.10.0,>=2.9.0rc0, but you have keras 2.11.0 which is incompatible.\n",
            "tensorflow 2.9.2 requires tensorboard<2.10,>=2.9, but you have tensorboard 2.11.0 which is incompatible.\n",
            "tensorflow 2.9.2 requires tensorflow-estimator<2.10.0,>=2.9.0rc0, but you have tensorflow-estimator 2.11.0 which is incompatible.\u001b[0m\n",
            "Successfully installed flatbuffers-22.12.6 keras-2.11.0 tensorboard-2.11.0 tensorflow-estimator-2.11.0 tensorflow-gpu-2.11.0\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "%cd PyTorch-WaveNet/"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Pnxd3734sotD",
        "outputId": "9ea8564f-d317-4788-9e25-417e1d9e08a8"
      },
      "execution_count": 7,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/content/PyTorch-WaveNet\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!python train.py --data_dir='/content/dataset'"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "yGLEVvDXs1bx",
        "outputId": "b657ac7f-ebc2-448c-db30-cc5b085a94d1"
      },
      "execution_count": 10,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "2022-12-22 07:21:55.209841: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA\n",
            "To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
            "2022-12-22 07:21:56.701037: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer.so.7'; dlerror: libnvinfer.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64\n",
            "2022-12-22 07:21:56.701216: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer_plugin.so.7'; dlerror: libnvinfer_plugin.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64\n",
            "2022-12-22 07:21:56.701240: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.\n",
            "Using default logdir: ./logdir/train/2022-12-22T07-22-00\n",
            "WARNING:tensorflow:From /content/PyTorch-WaveNet/wavenet/model.py:27: calling Constant.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
            "Instructions for updating:\n",
            "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
            "WARNING:tensorflow:From /content/PyTorch-WaveNet/wavenet/model.py:27: calling Constant.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
            "Instructions for updating:\n",
            "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
            "WARNING:tensorflow:From /usr/local/lib/python3.8/dist-packages/tensorflow/python/util/dispatch.py:1176: div (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.\n",
            "Instructions for updating:\n",
            "Deprecated in favor of operator or tf.math.divide.\n",
            "WARNING:tensorflow:From /usr/local/lib/python3.8/dist-packages/tensorflow/python/util/dispatch.py:1176: div (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.\n",
            "Instructions for updating:\n",
            "Deprecated in favor of operator or tf.math.divide.\n",
            "2022-12-22 07:22:24.483805: E tensorflow/compiler/xla/stream_executor/cuda/cuda_driver.cc:267] failed call to cuInit: CUDA_ERROR_NO_DEVICE: no CUDA-capable device is detected\n",
            "Trying to restore saved checkpoints from ./logdir/train/2022-12-22T07-22-00 ... No checkpoint found.\n",
            "WARNING:tensorflow:From train.py:289: start_queue_runners (from tensorflow.python.training.queue_runner_impl) is deprecated and will be removed in a future version.\n",
            "Instructions for updating:\n",
            "To construct input pipelines, use the `tf.data` module.\n",
            "WARNING:tensorflow:From train.py:289: start_queue_runners (from tensorflow.python.training.queue_runner_impl) is deprecated and will be removed in a future version.\n",
            "Instructions for updating:\n",
            "To construct input pipelines, use the `tf.data` module.\n",
            "WARNING:tensorflow:`tf.train.start_queue_runners()` was called when no queue runners were defined. You can safely remove the call to this deprecated function.\n",
            "WARNING:tensorflow:`tf.train.start_queue_runners()` was called when no queue runners were defined. You can safely remove the call to this deprecated function.\n",
            "files length: 3\n",
            "files length: 3\n",
            "files length: 3\n",
            "files length: 3\n",
            "files length: 3\n",
            "files length: 3\n",
            "files length: 3\n",
            "files length: 3\n",
            "files length: 3\n",
            "files length: 3\n",
            "files length: 3\n",
            "files length: 3\n",
            "step 0 - loss = 5.633, (50.856 sec/step)\n",
            "Storing checkpoint to ./logdir/train/2022-12-22T07-22-00 ...WARNING:tensorflow:Issue encountered when serializing variables.\n",
            "Type is unsupported, or the types of the items don't match field type in CollectionDef. Note this is a warning and probably safe to ignore.\n",
            "'filter_bias' has type str, but expected one of: int, long, bool\n",
            "WARNING:tensorflow:Issue encountered when serializing variables.\n",
            "Type is unsupported, or the types of the items don't match field type in CollectionDef. Note this is a warning and probably safe to ignore.\n",
            "'filter_bias' has type str, but expected one of: int, long, bool\n",
            "WARNING:tensorflow:Issue encountered when serializing trainable_variables.\n",
            "Type is unsupported, or the types of the items don't match field type in CollectionDef. Note this is a warning and probably safe to ignore.\n",
            "'filter_bias' has type str, but expected one of: int, long, bool\n",
            "WARNING:tensorflow:Issue encountered when serializing trainable_variables.\n",
            "Type is unsupported, or the types of the items don't match field type in CollectionDef. Note this is a warning and probably safe to ignore.\n",
            "'filter_bias' has type str, but expected one of: int, long, bool\n",
            " Done.\n",
            "step 1 - loss = 5.575, (28.123 sec/step)\n",
            "step 2 - loss = 5.617, (25.840 sec/step)\n",
            "files length: 3\n",
            "step 3 - loss = 5.584, (27.753 sec/step)\n",
            "step 4 - loss = 5.445, (28.130 sec/step)\n",
            "step 5 - loss = 5.522, (27.873 sec/step)\n",
            "files length: 3\n",
            "step 6 - loss = 5.594, (26.825 sec/step)\n",
            "step 7 - loss = 5.567, (30.323 sec/step)\n",
            "step 8 - loss = 5.382, (26.151 sec/step)\n",
            "files length: 3\n",
            "step 9 - loss = 5.411, (28.779 sec/step)\n",
            "step 10 - loss = 5.372, (27.870 sec/step)\n",
            "step 11 - loss = 5.360, (28.705 sec/step)\n",
            "files length: 3\n",
            "step 12 - loss = 5.341, (29.273 sec/step)\n",
            "step 13 - loss = 5.349, (37.824 sec/step)\n",
            "step 14 - loss = 5.297, (26.729 sec/step)\n",
            "files length: 3\n",
            "step 15 - loss = 5.521, (28.168 sec/step)\n",
            "step 16 - loss = 5.327, (25.533 sec/step)\n",
            "step 17 - loss = 5.314, (29.080 sec/step)\n",
            "files length: 3\n",
            "step 18 - loss = 5.297, (28.251 sec/step)\n",
            "step 19 - loss = 5.274, (26.999 sec/step)\n",
            "step 20 - loss = 5.473, (31.459 sec/step)\n",
            "files length: 3\n",
            "step 21 - loss = 5.452, (27.560 sec/step)\n",
            "step 22 - loss = 5.420, (26.711 sec/step)\n",
            "step 23 - loss = 5.323, (27.155 sec/step)\n",
            "files length: 3\n",
            "step 24 - loss = 5.227, (28.039 sec/step)\n",
            "step 25 - loss = 5.221, (29.165 sec/step)\n",
            "step 26 - loss = 5.210, (26.786 sec/step)\n",
            "files length: 3\n",
            "step 27 - loss = 5.198, (27.046 sec/step)\n",
            "step 28 - loss = 5.404, (29.328 sec/step)\n",
            "step 29 - loss = 5.402, (26.414 sec/step)\n",
            "files length: 3\n",
            "step 30 - loss = 5.301, (25.556 sec/step)\n",
            "step 31 - loss = 5.371, (25.408 sec/step)\n",
            "step 32 - loss = 5.302, (26.561 sec/step)\n",
            "files length: 3\n",
            "step 33 - loss = 5.300, (26.014 sec/step)\n",
            "step 34 - loss = 5.293, (25.464 sec/step)\n",
            "step 35 - loss = 5.356, (26.736 sec/step)\n",
            "files length: 3\n",
            "step 36 - loss = 5.344, (26.332 sec/step)\n",
            "step 37 - loss = 5.188, (24.968 sec/step)\n",
            "step 38 - loss = 5.291, (24.553 sec/step)\n",
            "files length: 3\n",
            "step 39 - loss = 5.289, (25.807 sec/step)\n",
            "step 40 - loss = 5.314, (24.172 sec/step)\n",
            "step 41 - loss = 5.274, (25.547 sec/step)\n",
            "files length: 3\n",
            "step 42 - loss = 5.311, (25.516 sec/step)\n",
            "step 43 - loss = 5.264, (27.401 sec/step)\n",
            "step 44 - loss = 5.259, (25.559 sec/step)\n",
            "files length: 3\n",
            "step 45 - loss = 5.174, (23.382 sec/step)\n",
            "step 46 - loss = 5.321, (25.088 sec/step)\n",
            "step 47 - loss = 5.155, (24.389 sec/step)\n",
            "files length: 3\n",
            "step 48 - loss = 5.144, (27.248 sec/step)\n",
            "step 49 - loss = 5.234, (25.846 sec/step)\n",
            "step 50 - loss = 5.221, (24.545 sec/step)\n",
            "Storing checkpoint to ./logdir/train/2022-12-22T07-22-00 ...WARNING:tensorflow:Issue encountered when serializing variables.\n",
            "Type is unsupported, or the types of the items don't match field type in CollectionDef. Note this is a warning and probably safe to ignore.\n",
            "'filter_bias' has type str, but expected one of: int, long, bool\n",
            "WARNING:tensorflow:Issue encountered when serializing variables.\n",
            "Type is unsupported, or the types of the items don't match field type in CollectionDef. Note this is a warning and probably safe to ignore.\n",
            "'filter_bias' has type str, but expected one of: int, long, bool\n",
            "WARNING:tensorflow:Issue encountered when serializing trainable_variables.\n",
            "Type is unsupported, or the types of the items don't match field type in CollectionDef. Note this is a warning and probably safe to ignore.\n",
            "'filter_bias' has type str, but expected one of: int, long, bool\n",
            "WARNING:tensorflow:Issue encountered when serializing trainable_variables.\n",
            "Type is unsupported, or the types of the items don't match field type in CollectionDef. Note this is a warning and probably safe to ignore.\n",
            "'filter_bias' has type str, but expected one of: int, long, bool\n",
            " Done.\n",
            "files length: 3\n",
            "step 51 - loss = 5.100, (27.780 sec/step)\n",
            "step 52 - loss = 5.188, (24.897 sec/step)\n",
            "step 53 - loss = 5.175, (25.499 sec/step)\n",
            "files length: 3\n",
            "step 54 - loss = 5.159, (25.801 sec/step)\n",
            "step 55 - loss = 5.054, (23.886 sec/step)\n",
            "step 56 - loss = 5.474, (26.686 sec/step)\n",
            "files length: 3\n",
            "step 57 - loss = 5.140, (27.670 sec/step)\n",
            "step 58 - loss = 5.121, (29.154 sec/step)\n",
            "step 59 - loss = 4.982, (29.415 sec/step)\n",
            "files length: 3\n",
            "step 60 - loss = 4.967, (25.092 sec/step)\n",
            "step 61 - loss = 4.932, (26.409 sec/step)\n",
            "step 62 - loss = 5.358, (25.632 sec/step)\n",
            "files length: 3\n",
            "step 63 - loss = 5.304, (26.258 sec/step)\n",
            "step 64 - loss = 5.132, (26.061 sec/step)\n",
            "step 65 - loss = 5.207, (25.279 sec/step)\n",
            "files length: 3\n",
            "step 66 - loss = 5.158, (29.078 sec/step)\n",
            "step 67 - loss = 4.777, (24.782 sec/step)\n",
            "step 68 - loss = 5.101, (25.378 sec/step)\n",
            "files length: 3\n",
            "step 69 - loss = 5.030, (26.148 sec/step)\n",
            "step 70 - loss = 4.991, (26.086 sec/step)\n",
            "step 71 - loss = 5.075, (27.395 sec/step)\n",
            "files length: 3\n",
            "step 72 - loss = 4.615, (28.215 sec/step)\n",
            "step 73 - loss = 4.855, (26.430 sec/step)\n",
            "step 74 - loss = 4.554, (28.229 sec/step)\n",
            "files length: 3\n",
            "step 75 - loss = 5.060, (25.216 sec/step)\n",
            "step 76 - loss = 4.985, (25.662 sec/step)\n",
            "step 77 - loss = 4.730, (24.998 sec/step)\n",
            "files length: 3\n",
            "step 78 - loss = 4.987, (25.463 sec/step)\n",
            "step 79 - loss = 4.675, (25.153 sec/step)\n",
            "step 80 - loss = 4.565, (26.048 sec/step)\n",
            "files length: 3\n",
            "step 81 - loss = 4.603, (26.550 sec/step)\n",
            "step 82 - loss = 4.562, (29.573 sec/step)\n",
            "step 83 - loss = 4.925, (29.199 sec/step)\n",
            "files length: 3\n",
            "step 84 - loss = 4.911, (28.744 sec/step)\n",
            "step 85 - loss = 4.306, (29.147 sec/step)\n",
            "step 86 - loss = 4.500, (27.705 sec/step)\n",
            "files length: 3\n",
            "step 87 - loss = 4.455, (27.704 sec/step)\n",
            "step 88 - loss = 4.425, (28.629 sec/step)\n",
            "step 89 - loss = 4.270, (30.141 sec/step)\n",
            "files length: 3\n",
            "step 90 - loss = 4.836, (28.006 sec/step)\n",
            "step 91 - loss = 4.389, (27.944 sec/step)\n",
            "step 92 - loss = 4.098, (27.477 sec/step)\n",
            "files length: 3\n",
            "step 93 - loss = 4.297, (27.482 sec/step)\n",
            "step 94 - loss = 4.063, (34.491 sec/step)\n",
            "step 95 - loss = 4.026, (31.012 sec/step)\n",
            "files length: 3\n",
            "step 96 - loss = 3.969, (28.604 sec/step)\n",
            "step 97 - loss = 4.186, (27.581 sec/step)\n",
            "step 98 - loss = 5.001, (29.857 sec/step)\n",
            "files length: 3\n",
            "step 99 - loss = 3.875, (26.668 sec/step)\n",
            "step 100 - loss = 4.848, (27.448 sec/step)\n",
            "Storing checkpoint to ./logdir/train/2022-12-22T07-22-00 ...WARNING:tensorflow:Issue encountered when serializing variables.\n",
            "Type is unsupported, or the types of the items don't match field type in CollectionDef. Note this is a warning and probably safe to ignore.\n",
            "'filter_bias' has type str, but expected one of: int, long, bool\n",
            "WARNING:tensorflow:Issue encountered when serializing variables.\n",
            "Type is unsupported, or the types of the items don't match field type in CollectionDef. Note this is a warning and probably safe to ignore.\n",
            "'filter_bias' has type str, but expected one of: int, long, bool\n",
            "WARNING:tensorflow:Issue encountered when serializing trainable_variables.\n",
            "Type is unsupported, or the types of the items don't match field type in CollectionDef. Note this is a warning and probably safe to ignore.\n",
            "'filter_bias' has type str, but expected one of: int, long, bool\n",
            "WARNING:tensorflow:Issue encountered when serializing trainable_variables.\n",
            "Type is unsupported, or the types of the items don't match field type in CollectionDef. Note this is a warning and probably safe to ignore.\n",
            "'filter_bias' has type str, but expected one of: int, long, bool\n",
            " Done.\n",
            "step 101 - loss = 4.801, (28.479 sec/step)\n",
            "files length: 3\n",
            "step 102 - loss = 4.740, (29.916 sec/step)\n",
            "step 103 - loss = 4.171, (27.765 sec/step)\n",
            "step 104 - loss = 3.998, (27.381 sec/step)\n",
            "files length: 3\n",
            "step 105 - loss = 4.098, (28.676 sec/step)\n",
            "step 106 - loss = 3.922, (28.729 sec/step)\n",
            "step 107 - loss = 3.920, (26.648 sec/step)\n",
            "files length: 3\n",
            "step 108 - loss = 3.866, (27.611 sec/step)\n",
            "step 109 - loss = 4.022, (30.838 sec/step)\n",
            "step 110 - loss = 3.974, (28.319 sec/step)\n",
            "files length: 3\n",
            "step 111 - loss = 4.784, (28.081 sec/step)\n",
            "step 112 - loss = 3.788, (27.459 sec/step)\n",
            "step 113 - loss = 3.971, (29.274 sec/step)\n",
            "files length: 3\n",
            "step 114 - loss = 3.931, (28.641 sec/step)\n",
            "step 115 - loss = 3.893, (31.922 sec/step)\n",
            "step 116 - loss = 4.624, (31.510 sec/step)\n",
            "files length: 3\n",
            "step 117 - loss = 3.812, (27.690 sec/step)\n",
            "step 118 - loss = 4.571, (27.776 sec/step)\n",
            "step 119 - loss = 4.552, (27.659 sec/step)\n",
            "files length: 3\n",
            "step 120 - loss = 4.540, (27.985 sec/step)\n",
            "step 121 - loss = 3.719, (28.510 sec/step)\n",
            "step 122 - loss = 4.499, (27.381 sec/step)\n",
            "files length: 3\n",
            "step 123 - loss = 3.951, (30.169 sec/step)\n",
            "step 124 - loss = 4.474, (27.038 sec/step)\n",
            "step 125 - loss = 3.869, (26.097 sec/step)\n",
            "files length: 3\n",
            "step 126 - loss = 4.446, (27.202 sec/step)\n",
            "step 127 - loss = 4.079, (27.130 sec/step)\n",
            "step 128 - loss = 3.618, (26.736 sec/step)\n",
            "files length: 3\n",
            "step 129 - loss = 3.661, (26.451 sec/step)\n",
            "step 130 - loss = 3.649, (27.838 sec/step)\n",
            "step 131 - loss = 4.394, (29.689 sec/step)\n",
            "files length: 3\n",
            "step 132 - loss = 4.398, (27.900 sec/step)\n",
            "step 133 - loss = 3.653, (27.255 sec/step)\n",
            "step 134 - loss = 4.365, (26.286 sec/step)\n",
            "files length: 3\n",
            "step 135 - loss = 4.042, (26.063 sec/step)\n",
            "step 136 - loss = 3.546, (26.305 sec/step)\n",
            "step 137 - loss = 3.505, (30.559 sec/step)\n",
            "files length: 3\n",
            "step 138 - loss = 4.336, (29.654 sec/step)\n",
            "step 139 - loss = 3.953, (26.536 sec/step)\n",
            "step 140 - loss = 3.455, (27.083 sec/step)\n",
            "files length: 3\n",
            "step 141 - loss = 3.492, (26.901 sec/step)\n",
            "step 142 - loss = 3.727, (27.551 sec/step)\n",
            "step 143 - loss = 3.651, (25.839 sec/step)\n",
            "files length: 3\n",
            "step 144 - loss = 3.396, (26.818 sec/step)\n",
            "step 145 - loss = 3.368, (28.823 sec/step)\n",
            "step 146 - loss = 4.343, (24.811 sec/step)\n",
            "files length: 3\n",
            "step 147 - loss = 3.371, (26.703 sec/step)\n",
            "step 148 - loss = 3.344, (26.075 sec/step)\n",
            "step 149 - loss = 3.679, (25.833 sec/step)\n",
            "files length: 3\n",
            "step 150 - loss = 3.572, (26.308 sec/step)\n",
            "Storing checkpoint to ./logdir/train/2022-12-22T07-22-00 ...WARNING:tensorflow:Issue encountered when serializing variables.\n",
            "Type is unsupported, or the types of the items don't match field type in CollectionDef. Note this is a warning and probably safe to ignore.\n",
            "'filter_bias' has type str, but expected one of: int, long, bool\n",
            "WARNING:tensorflow:Issue encountered when serializing variables.\n",
            "Type is unsupported, or the types of the items don't match field type in CollectionDef. Note this is a warning and probably safe to ignore.\n",
            "'filter_bias' has type str, but expected one of: int, long, bool\n",
            "WARNING:tensorflow:Issue encountered when serializing trainable_variables.\n",
            "Type is unsupported, or the types of the items don't match field type in CollectionDef. Note this is a warning and probably safe to ignore.\n",
            "'filter_bias' has type str, but expected one of: int, long, bool\n",
            "WARNING:tensorflow:Issue encountered when serializing trainable_variables.\n",
            "Type is unsupported, or the types of the items don't match field type in CollectionDef. Note this is a warning and probably safe to ignore.\n",
            "'filter_bias' has type str, but expected one of: int, long, bool\n",
            " Done.\n",
            "step 151 - loss = 3.298, (26.193 sec/step)\n",
            "step 152 - loss = 3.603, (24.264 sec/step)\n",
            "files length: 3\n",
            "step 153 - loss = 4.320, (28.714 sec/step)\n",
            "step 154 - loss = 3.766, (25.445 sec/step)\n",
            "step 155 - loss = 4.275, (25.633 sec/step)\n",
            "files length: 3\n",
            "step 156 - loss = 3.547, (27.630 sec/step)\n",
            "step 157 - loss = 4.222, (25.789 sec/step)\n",
            "step 158 - loss = 3.274, (25.545 sec/step)\n",
            "files length: 3\n",
            "step 159 - loss = 4.193, (27.207 sec/step)\n",
            "step 160 - loss = 3.261, (31.989 sec/step)\n",
            "step 161 - loss = 4.160, (28.236 sec/step)\n",
            "files length: 3\n",
            "step 162 - loss = 3.577, (25.521 sec/step)\n",
            "step 163 - loss = 3.227, (26.018 sec/step)\n",
            "step 164 - loss = 4.130, (25.697 sec/step)\n",
            "files length: 3\n",
            "step 165 - loss = 4.126, (26.783 sec/step)\n",
            "step 166 - loss = 3.513, (26.996 sec/step)\n",
            "step 167 - loss = 3.419, (26.171 sec/step)\n",
            "files length: 3\n",
            "step 168 - loss = 4.106, (28.443 sec/step)\n",
            "step 169 - loss = 4.104, (25.869 sec/step)\n",
            "step 170 - loss = 3.332, (25.548 sec/step)\n",
            "files length: 3\n",
            "step 171 - loss = 3.504, (24.998 sec/step)\n",
            "step 172 - loss = 4.054, (25.106 sec/step)\n",
            "step 173 - loss = 3.384, (24.921 sec/step)\n",
            "files length: 3\n",
            "step 174 - loss = 3.725, (25.426 sec/step)\n",
            "step 175 - loss = 3.399, (28.872 sec/step)\n",
            "step 176 - loss = 3.366, (29.747 sec/step)\n",
            "files length: 3\n",
            "step 177 - loss = 4.095, (29.700 sec/step)\n",
            "step 178 - loss = 3.458, (29.088 sec/step)\n",
            "step 179 - loss = 3.754, (29.548 sec/step)\n",
            "files length: 3\n",
            "step 180 - loss = 3.257, (26.873 sec/step)\n",
            "step 181 - loss = 4.025, (27.626 sec/step)\n",
            "step 182 - loss = 3.342, (31.758 sec/step)\n",
            "files length: 3\n",
            "step 183 - loss = 4.025, (29.590 sec/step)\n",
            "step 184 - loss = 4.064, (26.395 sec/step)\n",
            "step 185 - loss = 3.270, (26.520 sec/step)\n",
            "files length: 3\n",
            "step 186 - loss = 3.968, (26.789 sec/step)\n",
            "step 187 - loss = 3.232, (26.764 sec/step)\n",
            "step 188 - loss = 3.979, (27.661 sec/step)\n",
            "files length: 3\n",
            "step 189 - loss = 4.182, (29.372 sec/step)\n",
            "step 190 - loss = 3.773, (26.853 sec/step)\n",
            "step 191 - loss = 3.333, (26.849 sec/step)\n",
            "files length: 3\n",
            "step 192 - loss = 3.525, (26.030 sec/step)\n",
            "step 193 - loss = 3.327, (27.021 sec/step)\n",
            "step 194 - loss = 4.079, (25.841 sec/step)\n",
            "files length: 3\n",
            "step 195 - loss = 3.287, (25.652 sec/step)\n",
            "step 196 - loss = 3.238, (25.939 sec/step)\n",
            "step 197 - loss = 3.672, (30.426 sec/step)\n",
            "files length: 3\n",
            "step 198 - loss = 3.250, (27.337 sec/step)\n",
            "step 199 - loss = 4.071, (25.979 sec/step)\n",
            "step 200 - loss = 4.001, (25.920 sec/step)\n",
            "Storing checkpoint to ./logdir/train/2022-12-22T07-22-00 ...WARNING:tensorflow:Issue encountered when serializing variables.\n",
            "Type is unsupported, or the types of the items don't match field type in CollectionDef. Note this is a warning and probably safe to ignore.\n",
            "'filter_bias' has type str, but expected one of: int, long, bool\n",
            "WARNING:tensorflow:Issue encountered when serializing variables.\n",
            "Type is unsupported, or the types of the items don't match field type in CollectionDef. Note this is a warning and probably safe to ignore.\n",
            "'filter_bias' has type str, but expected one of: int, long, bool\n",
            "WARNING:tensorflow:Issue encountered when serializing trainable_variables.\n",
            "Type is unsupported, or the types of the items don't match field type in CollectionDef. Note this is a warning and probably safe to ignore.\n",
            "'filter_bias' has type str, but expected one of: int, long, bool\n",
            "WARNING:tensorflow:Issue encountered when serializing trainable_variables.\n",
            "Type is unsupported, or the types of the items don't match field type in CollectionDef. Note this is a warning and probably safe to ignore.\n",
            "'filter_bias' has type str, but expected one of: int, long, bool\n",
            " Done.\n",
            "files length: 3\n",
            "step 201 - loss = 3.732, (26.678 sec/step)\n",
            "step 202 - loss = 4.091, (25.328 sec/step)\n",
            "step 203 - loss = 3.416, (24.763 sec/step)\n",
            "files length: 3\n",
            "step 204 - loss = 3.967, (28.324 sec/step)\n",
            "step 205 - loss = 4.009, (26.532 sec/step)\n",
            "step 206 - loss = 3.316, (26.095 sec/step)\n",
            "files length: 3\n",
            "step 207 - loss = 3.631, (25.633 sec/step)\n",
            "step 208 - loss = 3.288, (27.263 sec/step)\n",
            "step 209 - loss = 3.958, (26.403 sec/step)\n",
            "files length: 3\n",
            "step 210 - loss = 3.749, (27.028 sec/step)\n",
            "step 211 - loss = 3.140, (29.970 sec/step)\n",
            "step 212 - loss = 3.890, (30.529 sec/step)\n",
            "files length: 3\n",
            "step 213 - loss = 4.195, (26.989 sec/step)\n",
            "step 214 - loss = 3.696, (27.735 sec/step)\n",
            "step 215 - loss = 3.859, (27.990 sec/step)\n",
            "files length: 3\n",
            "step 216 - loss = 3.752, (27.356 sec/step)\n",
            "step 217 - loss = 3.419, (27.796 sec/step)\n",
            "step 218 - loss = 3.719, (26.814 sec/step)\n",
            "files length: 3\n",
            "step 219 - loss = 3.192, (28.763 sec/step)\n",
            "step 220 - loss = 3.854, (27.830 sec/step)\n",
            "\n",
            "Storing checkpoint to ./logdir/train/2022-12-22T07-22-00 ...WARNING:tensorflow:From /usr/local/lib/python3.8/dist-packages/tensorflow/python/training/saver.py:1064: remove_checkpoint (from tensorflow.python.checkpoint.checkpoint_management) is deprecated and will be removed in a future version.\n",
            "Instructions for updating:\n",
            "Use standard file APIs to delete files with this prefix.\n",
            "WARNING:tensorflow:From /usr/local/lib/python3.8/dist-packages/tensorflow/python/training/saver.py:1064: remove_checkpoint (from tensorflow.python.checkpoint.checkpoint_management) is deprecated and will be removed in a future version.\n",
            "Instructions for updating:\n",
            "Use standard file APIs to delete files with this prefix.\n",
            "WARNING:tensorflow:Issue encountered when serializing variables.\n",
            "Type is unsupported, or the types of the items don't match field type in CollectionDef. Note this is a warning and probably safe to ignore.\n",
            "'filter_bias' has type str, but expected one of: int, long, bool\n",
            "WARNING:tensorflow:Issue encountered when serializing variables.\n",
            "Type is unsupported, or the types of the items don't match field type in CollectionDef. Note this is a warning and probably safe to ignore.\n",
            "'filter_bias' has type str, but expected one of: int, long, bool\n",
            "WARNING:tensorflow:Issue encountered when serializing trainable_variables.\n",
            "Type is unsupported, or the types of the items don't match field type in CollectionDef. Note this is a warning and probably safe to ignore.\n",
            "'filter_bias' has type str, but expected one of: int, long, bool\n",
            "WARNING:tensorflow:Issue encountered when serializing trainable_variables.\n",
            "Type is unsupported, or the types of the items don't match field type in CollectionDef. Note this is a warning and probably safe to ignore.\n",
            "'filter_bias' has type str, but expected one of: int, long, bool\n",
            " Done.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!python generate.py --wav_out_path=generated.wav --save_every 2000 --samples 16000 /content/PyTorch-WaveNet/logdir/train/2022-12-22T07-22-00/model.ckpt-221\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "DcqyixNWxGol",
        "outputId": "c7ac47cd-071d-4942-9dee-c1871433b4ec"
      },
      "execution_count": 11,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "2022-12-22 09:05:49.253744: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA\n",
            "To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
            "2022-12-22 09:05:53.222631: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer.so.7'; dlerror: libnvinfer.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64\n",
            "2022-12-22 09:05:53.223022: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer_plugin.so.7'; dlerror: libnvinfer_plugin.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64\n",
            "2022-12-22 09:05:53.223050: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.\n",
            "2022-12-22 09:05:58.247676: E tensorflow/compiler/xla/stream_executor/cuda/cuda_driver.cc:267] failed call to cuInit: CUDA_ERROR_NO_DEVICE: no CUDA-capable device is detected\n",
            "WARNING:tensorflow:From /content/PyTorch-WaveNet/wavenet/model.py:27: calling Constant.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
            "Instructions for updating:\n",
            "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
            "WARNING:tensorflow:From /content/PyTorch-WaveNet/wavenet/model.py:27: calling Constant.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
            "Instructions for updating:\n",
            "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
            "Restoring model from /content/PyTorch-WaveNet/logdir/train/2022-12-22T07-22-00/model.ckpt-221\n",
            "Updated wav file at generated.wav\n",
            "Updated wav file at generated.wav\n",
            "Updated wav file at generated.wav\n",
            "Updated wav file at generated.wav\n",
            "Updated wav file at generated.wav\n",
            "Updated wav file at generated.wav\n",
            "Updated wav file at generated.wav\n",
            "Updated wav file at generated.wav\n",
            "\n",
            "Updated wav file at generated.wav\n",
            "Finished generating. The result can be viewed in TensorBoard.\n"
          ]
        }
      ]
    }
  ]
}